src/xunit.analyzers/Utility/Descriptors.xUnit1xxx.cs

```diff
@@ -402,9 +402,23 @@ public static partial class Descriptors
 			"Constructor '{0}' must be public to be used on a test method."
 		);
 
-	// Placeholder for rule X1044
+	public static DiagnosticDescriptor X1044_TheoryDataTypeArgumentsShouldBeSerializable { get; } =
+		Diagnostic(
+			"xUnit1044",
+			"TheoryData type arguments should be serializable",
+			Usage,
+			Info,
+			"The type argument {0} is not serializable, which will cause Test Explorer to not enumerate individual data rows. Consider using a type that is known to be serializable."
+		);
 
-	// Placeholder for rule X1045
+	public static DiagnosticDescriptor X1045_TheoryDataTypeArgumentsShouldBeDefinitelySerializable { get; } =
+		Diagnostic(
+			"xUnit1045",
+			"TheoryData type arguments should be definitely serializable",
+			Usage,
+			Info,
+			"The type argument {0} might not be serializable, which may cause Test Explorer to not enumerate individual data rows. Consider using a type that is known to be serializable."
+		);
```

Comment:
Do you have any alternative suggestions for titles and property names for these two diagnostics?

I tried to come up with a pair of titles that clearly distinguished between the two without excessive wordiness or double negatives. This is what I came up with after trying out some options, but I'm not sure I'm happy with it.

Reply:
Yeah, this one's weird. It's almost like it's better to invert the naming to be the problem rather than the solution because "Should" is applies to both.

Maybe:

`X1044_TheoryDataTypeArgumentMightNotBeSerializable`
`X1045_TheoryDataTypeArgumentIsNotSerializable`

With titles:

`TheoryData type arguments might not be serializable`
`TheoryData type arguments are not serializable`

Thoughts?

Reply:
I went through the existing diagnostics and did an inventory of the language used, to see whether there is a precedent for this.

Most use standard prescriptive language: `must (not)`, `should (not)`, `can(not)`, `is recommended`.

A few use imperative voice: `ensure`, `do not`, `use`.

Only one uses descriptive language in the property name, describing the problem rather than the solution: `X1028_TestMethodHasInvalidReturnType` (though the title is phrased prescriptively: `Test method must have valid return type`).

I like your suggestions, as they are clear and simple.

Here's another option, using imperative voice with `avoid using`:

`Avoid using TheoryData type arguments that are not serializable`
`Avoid using TheoryData type arguments that might not be serializable`

I suppose mine are slightly more consistent with existing style, but also slightly more wordy. I'm happy to go with whichever you prefer.

Reply:
By the way, did you intend to switch the diagnostic ids? I currently have X1044 as the "is not serializable" diagnostic and X1045 as the "might not be serializable" one.

Reply:
No, that was unintentional.

Reply:
I like your suggestions. 👍🏼

Reply:
Ok, sounds good.